.semaphore/semaphore.yml

```diff
@@ -0,0 +1,88 @@
+---
+version: "v1.0"
+name: "Vaultenv CI Pipeline"
+
+agent:
+  machine:
+    type: "e1-standard-2"
+    os_image: "ubuntu2004"
+
+blocks:
+  - name: "Run checks"
+    task:
+      prologue:
+        commands:
+          - "checkout"
+
+          # Haskell requires libgmp for big integers.
+          - "sudo apt install libgmp-dev"
+
+          # TODO: verify installation script
+          # Install Stack from source
+          - "mkdir -p ~/.local/bin"
+          - "export PATH=$HOME/.local/bin:$PATH"
+          - "wget https://github.com/commercialhaskell/stack/releases/download/v2.7.3/stack-2.7.3-linux-x86_64.tar.gz"
+          - "sha256sum --check .semaphore/install-stack.2.7.3.sha256"
+          - "cat stack-2.7.3-linux-x86_64.tar.gz | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'"
+
+          # restore stack cache
+          - "cache restore stack-"
+
+      jobs:
+        - name: "Run tests for client"
+          commands:
+            - "(cd client-haskell && stack setup -j2 --no-terminal)"
+            - "(cd client-haskell && stack build -j2 --no-terminal)"
+            - "(cd client-haskell && stack test  -j2 --no-terminal)"
+
+        - name: "Run tests for server"
+          commands:
+            - "(cd server && stack setup -j2 --no-terminal)"
+            - "(cd server && stack build -j2 --no-terminal)"
+            - "(cd server && stack test  -j2 --no-terminal)"
+
+      epilogue:
+        always:
+          commands:
+            # Store a copy of the stack installation folder
+            - "cache store stack-$(date -u -Idate) ~/.stack"
+
+  - name: "Build"
```

Comment:
It seems that in semaphore, there's a dependency from this step to the previous one, but those are actually independent, so we could speed up the build a bit more.

This second step ensures that the nix build keeps working (which is how we use icepeak internally), while the other step makes sure that the pure stack build keeps working (which is still a lot more common in the Haskell world).

Reply:
I'm unsure why Semaphore thinks that there is a dependency, I'll see what I can do

Reply:
Fixed by making explicit that there are no dependencies :)